producers/prometheus/prometheus_test.go

```diff
@@ -17,16 +17,88 @@
 package prometheus
 
 import (
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
 	"testing"
+	"time"
 
 	"github.com/dcos/dcos-metrics/producers"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestNew(t *testing.T) {
 	Convey("The New method should return a producer and a channel", t, func() {
-		pi, pc := New()
+		pi, pc := New(Config{})
 		So(pi, ShouldHaveSameTypeAs, &promProducer{})
 		So(pc, ShouldHaveSameTypeAs, make(chan producers.MetricsMessage))
 	})
 }
+
+func TestRun(t *testing.T) {
+	Convey("Should create a Prometheus server on the configured port", t, func() {
+		// Ensure the environment is clean
+		if err := os.Unsetenv("LISTEN_PID"); err != nil {
+			panic(err)
+		}
+		if err := os.Unsetenv("LISTEN_FDS"); err != nil {
+			panic(err)
+		}
+
+		port, err := getEphemeralPort()
+		if err != nil {
+			panic(err)
+		}
+
+		p, metricsChan := New(Config{Port: port})
+		go p.Run()
+		time.Sleep(1 * time.Second)
+
+		// The web server should be listening on the configured TCP port
+		_, err = net.Dial("tcp", net.JoinHostPort("localhost", strconv.Itoa(port)))
+		So(err, ShouldBeNil)
+
+		// feed in some fake data
+		dps := []producers.Datapoint{
+			producers.Datapoint{
+				Name:  "datapoint-one",
+				Value: 123.456,
+			},
+		}
+		metricsChan <- producers.MetricsMessage{
+			Name:       "some-message",
+			Datapoints: dps,
+			Timestamp:  time.Now().UTC().Unix(),
+		}
+		time.Sleep(250 * time.Millisecond)
+
+		resp, err := http.Get("http://" + net.JoinHostPort("localhost", strconv.Itoa(port)) + "/metrics")
+		So(err, ShouldBeNil)
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		So(err, ShouldBeNil)
+
+		So(string(body), ShouldContainSubstring, "datapoint_one")
```

Comment:
Maybe leave a marker or TODO here that we should verify the response format a little more exactly.